### ISSUE TITLE
Handling of @Id and @Version fields in AutoCrud could be improved

### ISSUE THREAD

**Issue by rbrki07** [vaadin/hilla#2809]
### Describe the bug

When you have a JPA entity with a field annotated with `@Id` or `@Version` and you use this entity in the AutoCrud component, you likely won't be able to submit the form of the AutoCrud component. You can fill the form and press submit, but nothing will happen, and you won't see any errors in the frontend or backend.

Hilla validates these fields already on the client, but the error is never shown because AutoCrud / AutoForm hide fields for id and version automatically. So there are client-side validation errors, but they are not visible. The values for id and version fields are generated only on the server after saving an entity.

A workaround is to add the `@com.vaadin.hilla.Nullable` annotation to the id and/or version field of the JPA entity, but this feels strange. From a developer perspective, I want this fields to be not empty nor null, and I know, that JPA will take care of this.

### Expected-behavior

Using the annotations `@Id` and `@Version` is quite common. As a developer, I want to use entities with these annotations in the AutoCrud component with a good developer experience. This could mean:
* Getting an error/warning when the validation of the hidden id and/or version field in the AutoCrud components fails. This could be a `console.error` or a notification or a popup/popover.
* AutoCrud could skip validation for fields that are hidden per default. 

### Reproduction

* Create a new Hilla project using start.vaadin.com or Vaadin CLI
* Add an entity with id and/or version field and `@Id` and/or `@Version` annotations
* Create a JpaRepository and CrudRepositoryService for that entity
* Create a view, add the AutoCrud component to that view and configure the component to use the generated model and service of that entity.
* Try to add a new item with the form component of the AutoCrud component

### System Info

Vaadin 24.5.0.beta6

**Comment by rbrki07** [comment]
Origin: https://vaadin.com/forum/t/hilla-autocruds-submmit-button-wont-persist-data-on-database/166588

**Comment by platosha** [comment]
See #2217 specifically about the errors from hidden fields. Besides that, we could assume `@Nullable` for such `@Id` and `@Version` fields, allowing the value to be undefined on the client side.

**Comment by rbrki07** [comment]
I just realized that I was able to use entities with id and/or version fields together with AutoCrud and AutoForm in a previous app using Vaadin 24.4. Did this behaviour change on purpose?

**Comment by rbrki07** [comment]
I think I need to add some more context. I detected a typo in my example code
```java
@Id
@GeneratedValue(strategy = GenerationType.IDENTITY)
private long id;
```

For some reason I don't understand, I used `long` instead of `Long` which results in the behaviour described above.

Using `private long id` means that I was not able to submit a AutoCrud or AutoForm component, because Hilla's client-side validation prevents the form from being submitted to the backend.

Using `private Long id` works just fine, which means that I was able to submit a AutoCrud or AutoForm component.
